packages/react-storage/src/components/StorageBrowser/Views/LocationDetailView/Controls/DataTable.tsx

```diff
@@ -0,0 +1,275 @@
+import React from 'react';
+
+import { humanFileSize } from '@aws-amplify/ui';
+
+import {
+  DataTable,
+  TABLE_DATA_BUTTON_CLASS,
+  TABLE_DATA_CLASS_NAME,
+  TABLE_DATA_ICON_CLASS,
+  TABLE_DATA_TEXT_CLASS_NAME,
+  TABLE_HEADER_BUTTON_CLASS_NAME,
+  TABLE_HEADER_CLASS_NAME,
+} from '../../../components/DataTable';
+import { useControl } from '../../../context/controls';
+import { useAction } from '../../../context/actions';
+import { LocationItem } from '../../../context/types';
+import {
+  compareDates,
+  compareNumbers,
+  compareStrings,
+} from '../../../context/controls/Table';
+import {
+  ButtonElement,
+  IconElement,
+  SpanElement,
+} from '../../../context/elements';
+import { DownloadControl } from '../../Controls';
+import { getFileExtension } from '../../utils';
+
+export type SortDirection = 'ascending' | 'descending' | 'none';
+
+export type SortState = {
+  selection: string;
+  direction: SortDirection;
+};
+
+type Column<T> = {
+  key: keyof T;
+  header: string;
+  compareFn?: (a: T, b: T) => number;
+};
+
+type DataTableLocationItems = {
+  key: string;
+  fileExt: string;
+  lastModified: Date | undefined;
+  size: number | undefined;
+  type: string;
+};
+
+// Define the columns for the data table
+const columnData: Column<DataTableLocationItems>[] = [
+  {
+    key: 'key',
+    header: 'Name',
+    compareFn: (a, b) => compareStrings(a.key, b.key),
+  },
+  {
+    key: 'fileExt',
+    header: 'Type',
+    compareFn: (a, b) => compareStrings(a.fileExt, b.fileExt),
+  },
+  {
+    key: 'lastModified',
+    header: 'Last Modified',
+    compareFn: (a, b) => compareDates(a.lastModified!, b.lastModified!),
+  },
+  {
+    key: 'size',
+    header: 'Size',
+    compareFn: (a, b) => compareNumbers(a.size!, b.size!),
+  },
+];
+
+// Generate a table header item
+const getColumnItem = ({
+  columnData,
+  selection,
+  direction,
+  onTableHeaderClick,
+}: {
+  columnData: Column<DataTableLocationItems>;
+  selection: string;
+  direction: SortDirection;
+  onTableHeaderClick: (location: string) => void;
+}): Record<string, any> => {
+  const { key, header } = columnData;
+
+  return {
+    children: (
+      <ButtonElement variant="sort" className={TABLE_HEADER_BUTTON_CLASS_NAME}>
+        {header}
+        <IconElement
+          variant={
+            selection === key && direction !== 'none'
+              ? `sort-${direction}`
+              : 'sort-indeterminate'
+          }
+        />
+      </ButtonElement>
+    ),
+    key: `th_${key}`,
+    className: `${TABLE_HEADER_CLASS_NAME} ${TABLE_HEADER_CLASS_NAME}--${key}`,
+    onClick: () => onTableHeaderClick(key),
+    'aria-sort': selection === key ? direction : 'none',
+  };
+};
+
+// Generate the data for each row in the table
+const getLocationsItemData = ({
```

Comment:
Bit concerned abt the size of this utility, feels like there should be a separation of concerns